### ISSUE TITLE
Client-side Frameworks and Libraries

### ISSUE THREAD

**Issue by sumn2u** [sumn2u/learn-javascript#32]
It would be nice to have a separate chapter that describes client-side frameworks and libraries, as this would equip developers with the knowledge and skills needed to efficiently create modern, feature-rich web applications that meet user expectations and industry standards.

**Comment by kn1ghtm0nster** [comment]
I'm curious about this one! Is this chapter focusing on just JS frameworks like Vue, React, Angular etc. and how they operate or just a brief breakdown of how they interact with JS / TS as a whole?

**Comment by sumn2u** [comment]
> I'm curious about this one! Is this chapter focusing on just JS frameworks like Vue, React, Angular etc. and how they operate or just a brief breakdown of how they interact with JS / TS as a whole?

@kn1ghtm0nster  Yes, it focuses on state-of-the-art JS frameworks and explains their working mechanisms with references for deeper understanding.

**Comment by kn1ghtm0nster** [comment]
> 

Got it. So more of "here are some commonly used front-end frameworks that are built on top of JS" and then just a brief explanations of how they help build on top of JS with references as you've pointed out. I can for sure start taking a look at this since I was taught vanilla JS before using React and now learning Angular for work.

**Comment by sumn2u** [comment]
Cool! I'm glad to have you onboard. I'll assign this task to you. It'll be a learning experience for me too since I haven't worked with Angular in a while :)

**Comment by kn1ghtm0nster** [comment]
> Cool! I'm glad to have you onboard. I'll assign this task to you. It'll be a learning experience for me too since I haven't worked with Angular in a while :)

Sweet! Thanks for assigning this to me! I will start writing some stuff up. How did you want me to provide status updates?

**Comment by sumn2u** [comment]
@kn1ghtm0nster  You can create a draft pull request (PR) and add content. Once complete, convert it to a PR and assign me for review. For discussion, comment on the PR or write [here](https://github.com/sumn2u/learn-javascript/discussions).
